File: CSharpCodeBase/aiactions/ai_runawayfromtarget.lua
local Class = require 'utils.hump.class'
local AINode = require 'utils.ailib.ainode'

-- Move to enemy
local AI_RunawayFromTarget = Class({__includes=AINode, init = function (self, entity)
  AINode.init(self, entity)
  self.distance = self.entity.config:Get("scarySafeRadius")
end})

function AI_RunawayFromTarget:Visit()
  local target = self.parent.target
  if self.status == NODE_READY then
    self.status = NODE_RUNNING
  elseif self.status == NODE_RUNNING then
    if self.entity:GetSimpleDistance(target) > self.distance then
      self.status = NODE_SUCCESS
      self.entity.mover:Stop()
    else
      self.entity.mover:SetSpeed(self.entity.config:Get("scaryRunSpeed"))
      self.entity.mover:SetInputFromVec(target:GetPosition(), false, self.entity.config:Get("scaryRunType"))
    end
  end
end

return AI_RunawayFromTarget

File: CSharpCodeBase/aiactions/ai_runawayfromtarget.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class ai_runawayfromtarget :ainode{
 // Move to enemy
 public void init (self, entity);
   AINode.init(self, entity);
   this.distance = this.entity.config:Get("scarySafeRadius");
 }})
 public void Visit(){
   var target = this.parent.target;
   if(this.status == NODE_READY  ){ 
     this.status = NODE_RUNNING;
   }else{if this.status == NODE_RUNNING  ){ 
     if(this.entity:GetSimpleDistance(target) > this.distance  ){ 
       this.status = NODE_SUCCESS;
       this.entity.mover:Stop();
     }else{
       this.entity.mover:SetSpeed(this.entity.config:Get("scaryRunSpeed"));
       this.entity.mover:SetInputFromVec(target:GetPosition(), false, this.entity.config:Get("scaryRunType"));
     }
   }
 }
 }}